### PR TITLE
fix dependency on social-auth-core

### LINKED ISSUE #5565: Remove `openidconnect` Extra from Social Auth Core, Removed Upstream
We can remove `openidconnect` extra from Socaial Auth Core as they ditched the dependency on `jose`: https://github.com/python-social-auth/social-core/pull/819

Related: https://github.com/python-social-auth/social-core/issues/895

Our line here: https://github.com/nautobot/nautobot/blob/af15204ee7dedc741593e7f039e0abcc3d8b346b/pyproject.toml#L133

Without doing it, when installing `nautobot[sso]` you will receive the error:

```
social-auth-core 4.5.3 does not provide the extra 'openidconnect'
```

### PR DESCRIPTION
**Pull request by cardoe**
Since 4.5.0 of social-auth-core it has not had an extras of 'openidconnect' so drop it from the dependencies.
see https://github.com/python-social-auth/social-core/issues/895 fixes #5565

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example App Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/5565.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+fixes dependency on social-auth-core by removing an extras related to openidconnect that no longer exists
```
Comment (glennmatthews) [inline changes/5565.fixed:1]:
```suggestion
Fixed optional dependency on `social-auth-core` by removing an extras related to openidconnect that no longer exists.
```

**Review by glennmatthews — APPROVED** [review]
Looks good, thanks! I'm a bit surprised that it didn't result in any change to `poetry.lock` but I tried it locally (`poetry add --optional social-auth-core~4.5.3 --extras saml`) and that seems to be the case.

**Comment by glennmatthews** [comment]
Spoke too soon :) I didn't notice the CI failures. Try `poetry add --optional social-auth-core~4.5.3 --extras saml` and see if that updates the poetry.lock to make CI happy.

File: changes/5565.fixed (hunk 1 above)
Comment (cardoe) [reply changes/5565.fixed]: Thanks.

**Comment by cardoe** [comment]
@glennmatthews should be rebased and pass cleanly

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**